### DISCUSSION TITLE
can I use react hook form with react-slate

### DISCUSSION THREAD

**Discussion by 0xMALVEE** [ianstormtaylor/slate#5613] [Q&A] [answered]
can I use react hook form with react-slate

**Comment by gerardnico** [comment, marked as answer]
I know that it's not a fresh answer but I was facing the same problem.

The answer that I found is:
* use [setValue](https://react-hook-form.com/docs/useform/setvalue)
* in a [onChange or the new onValueChange](https://docs.slatejs.org/libraries/slate-react/slate) callback

All the best

**Reply by 0xMALVEE** [reply]
perfect. I am also using the same way you suggested after digging a little.

**Reply by mikeycoxon** [reply]
is there a code example for any of this somewhere? so, sure use react-hook-form's setValue...

but on what? initialValue works, unless you do a browser refresh, so if not initialValue, then what?

**Reply by gerardnico** [reply]
Mike reread my answer. There are 2 points. The answer to your question is the second point.

You use react hook set value in a onchange slate editor callback.

I answer with a mobile on holiday so sorry no example code but I’m sure you will overcome it. 

Happy new year!
